File: src/cviz/timeline/commands/LoopCommand.java
package cviz.timeline.commands;

import cviz.ITimeline;
import cviz.LayerState;
import cviz.timeline.Trigger;
import se.svt.caspar.amcp.AmcpLayer;
import se.svt.caspar.producer.Video;

public class LoopCommand extends ICommand {
    public LoopCommand(int layerId) {
        super(layerId);
    }

    @Override
    public boolean execute(ITimeline timeline) {
        AmcpLayer layer = timeline.getLayer(getLayerId());

        try {
            layer.play();
            LayerState state = timeline.getLayerState(getLayerId());
            if(state == null)
                throw new Exception("Missing layer state for loop " + getLayerId());

            layer.loadBg(new Video(state.getVideoName()));

            Trigger t = Trigger.CreateLoop(getLayerId());
            timeline.getActiveTriggers().add(t);

            System.out.println("Looping: " + state.toString());

            return true;
        } catch (Exception e){
            System.err.println("Failed to execute command: " + e.getMessage());
            return false;
        }
    }

    @Override
    public String toString() {
        return "LoopCommand: " + getLayerId();
    }
}

File: CViz/Timeline/Command/LoopCommand.cs
using System;
using CViz.Timeline.Triggers;
using log4net;
using StilSoft.CasparCG.AmcpClient.Commands.Basic;

namespace CViz.Timeline.Command
{
    class LoopCommand : CommandBase
    {
        private static readonly ILog Log = LogManager.GetLogger(typeof(LoopCommand));

        public LoopCommand(int layerId) : base(layerId)
        {
        }

        public override bool Execute(ITimeline timeline)
        {
            try
            {
                new PlayCommand(timeline.ChannelNumber, LayerId).Execute(timeline.Client);
                LayerState state = timeline.GetLayerState(LayerId);
                if (state == null)
                    throw new Exception("Missing layer state for loop " + LayerId);

                new LoadBgCommand(timeline.ChannelNumber, LayerId, state.VideoName).Execute(timeline.Client);
                timeline.AddTrigger(new LoopTrigger(LayerId));
                
                Log.InfoFormat("Looping: {0}", state);

                return true;
            }
            catch (Exception e)
            {
                Log.ErrorFormat("Failed to execute command: {0}", e.Message);
                return false;
            }
        }

        public override string ToString()
        {
            return $"LoopCommand: {LayerId}";
        }
    }
}